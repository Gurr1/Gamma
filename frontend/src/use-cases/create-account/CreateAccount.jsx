import React, { Component } from "react";
import PropTypes from "prop-types";

import InputCid from "./views/input-cid";
import CreationOfAccountFinished from "./views/creation-of-account-finished";
import EmailHasBeenSent from "./views/email-has-been-sent";
import InputDataAndCode from "./views/input-data-and-code";

import GammaStepper from "../../common/elements/gamma-stepper";
import MapPathToStep from "../../common/declaratives/map-path-to-step";
import { Fill, Spacing } from "../../common-ui/layout";

import { DigitTranslations } from "@cthit/react-digit-components";
import translations from "./CreateAccount.translations.json";

class CreateAccount extends React.Component {
    constructor(props) {
        super();

        props.gammaLoadingFinished();
    }

    render() {
        const { location } = this.props;

        console.log(translations);

        return (
            <DigitTranslations
                translations={translations}
                uniquePath="CreateAccount"
                render={text => (
                    <Fill>
                        {console.log(text)}
                        <MapPathToStep
                            currentPath={location.pathname}
                            pathToStepMap={{
                                "/create-account": 0,
                                "/create-account/email-sent": 1,
                                "/create-account/input": 2,
                                "/create-account/finished": 3
                            }}
                            render={step => (
                                <GammaStepper
                                    activeStep={step}
                                    steps={[
                                        {
                                            text: text.SendCid,
                                            element: <InputCid />
                                        },
                                        {
                                            text: text.GetActivationCode,
                                            element: <EmailHasBeenSent />
                                        },
                                        {
                                            text: text.CreateAccount,
                                            element: <InputDataAndCode />
                                        }
                                    ]}
                                    finishedElement={
                                        <CreationOfAccountFinished />
                                    }
                                />
                            )}
                        />
                    </Fill>
                )}
            />
        );
    }
}

CreateAccount.propTypes = {
    location: PropTypes.object.isRequired
};

export default CreateAccount;
