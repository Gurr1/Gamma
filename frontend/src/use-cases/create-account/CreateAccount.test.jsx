import React from "react";
import { shallow } from "enzyme";
import CreateAccount from "./CreateAccount";

describe("<CreateAccount/>", () => {
  test("Shallow render of <CreateAccount/>", () => {
    const wrapper = shallow(
      <CreateAccount location={{ pathname: "/" }} text={{}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
