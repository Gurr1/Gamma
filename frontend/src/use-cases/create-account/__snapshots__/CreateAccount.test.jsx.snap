// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAccount/> Shallow render of <CreateAccount/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateAccount
    location={
        Object {
            "pathname": "/",
          }
    }
    text={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.div>
          <MapPathToStep
                    currentPath="/"
                    pathToStepMap={
                              Object {
                                        "/create-account": 0,
                                        "/create-account/email-sent": 1,
                                        "/create-account/finished": 3,
                                        "/create-account/input": 2,
                                      }
                    }
                    render={[Function]}
          />
</styled.div>,
        <styled.div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MapPathToStep
            currentPath="/"
            pathToStepMap={
                        Object {
                                    "/create-account": 0,
                                    "/create-account/email-sent": 1,
                                    "/create-account/finished": 3,
                                    "/create-account/input": 2,
                                  }
            }
            render={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPath": "/",
            "pathToStepMap": Object {
              "/create-account": 0,
              "/create-account/email-sent": 1,
              "/create-account/finished": 3,
              "/create-account/input": 2,
            },
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div>
            <MapPathToStep
                        currentPath="/"
                        pathToStepMap={
                                    Object {
                                                "/create-account": 0,
                                                "/create-account/email-sent": 1,
                                                "/create-account/finished": 3,
                                                "/create-account/input": 2,
                                              }
                        }
                        render={[Function]}
            />
</styled.div>,
          <styled.div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MapPathToStep
              currentPath="/"
              pathToStepMap={
                            Object {
                                          "/create-account": 0,
                                          "/create-account/email-sent": 1,
                                          "/create-account/finished": 3,
                                          "/create-account/input": 2,
                                        }
              }
              render={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentPath": "/",
              "pathToStepMap": Object {
                "/create-account": 0,
                "/create-account/email-sent": 1,
                "/create-account/finished": 3,
                "/create-account/input": 2,
              },
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
